### DISCUSSION TITLE
Helm chart 7.0.0-alpha1: Access with port-forward returns 404

### DISCUSSION THREAD

**Discussion by Stolz** [kubernetes/dashboard#8447] [Q&A]
I've installed the dashboard using the latest  official helm chart version (7.0.0-alpha1)  with these `values.yaml` file

```yml
app:
  security:
    networkPolicy:
      enabled: true
      ingressDenyAll: true
  ingress:
    enabled: false
metrics-server:
  enabled: false
metricsScraper:
  enabled: false
```

As you can see, metrics and ingress are disabled. I don't have an ingress controller installed in my cluster yet but [according to the official documentation](https://artifacthub.io/packages/helm/k8s-dashboard/kubernetes-dashboard#networkpolicy) I should still be able to access with `kubectl port-forward`.

The pods and services seem to be successfully created 
```
kubectl get pods -n kubernetes-dashboard
NAME                                                            READY   STATUS    RESTARTS   AGE
kubernetes-dashboard-api-6988dbb97d-7mzqr                       1/1     Running   0           1m
kubernetes-dashboard-cert-manager-76ff4cb656-s7wfw              1/1     Running   0           1m
kubernetes-dashboard-cert-manager-cainjector-55db97cfbb-7hjlh   1/1     Running   0           1m
kubernetes-dashboard-cert-manager-webhook-7b569977cb-2d7jt      1/1     Running   0           1m
kubernetes-dashboard-nginx-controller-65474fdcb8-4n6cp          1/1     Running   0           1m
kubernetes-dashboard-web-58fb55564c-cxkf4                       1/1     Running   0           1m

kubectl get svc -n kubernetes-dashboard
NAME                                              TYPE        CLUSTER-IP        EXTERNAL-IP   PORT(S)          AGE
kubernetes-dashboard-api                          ClusterIP   192.168.50.59     <none>        9000/TCP          2m
kubernetes-dashboard-cert-manager                 ClusterIP   192.168.220.8     <none>        9402/TCP          2m
kubernetes-dashboard-cert-manager-webhook         ClusterIP   192.168.15.227    <none>        443/TCP           2m
kubernetes-dashboard-metrics-scraper              ClusterIP   192.168.152.165   <none>        8000/TCP          2m
kubernetes-dashboard-nginx-controller             ClusterIP   192.168.179.192   <none>        80/TCP,443/TCP    2m
kubernetes-dashboard-nginx-controller-admission   ClusterIP   192.168.28.152    <none>        443/TCP           2m
kubernetes-dashboard-web                          ClusterIP   192.168.238.71    <none>        8000/TCP          2m
```

But when I try to access with port-forward I get a 404 error

```
kubectl -n kubernetes-dashboard port-forward svc/kubernetes-dashboard-nginx-controller 8443:443
Forwarding from 127.0.0.1:8443 -> 443
Handling connection for 8443
```
```
wget --no-check-certificate https://127.0.0.1:8443/
--2023-11-03 21:25:56--  https://127.0.0.1:8443/
Connecting to 127.0.0.1:8443... connected.
WARNING: cannot verify 127.0.0.1's certificate, issued by 'CN=Kubernetes Ingress Controller Fake Certificate,O=Acme Co':
  Self-signed certificate encountered.
    WARNING: certificate common name 'Kubernetes Ingress Controller Fake Certificate' doesn't match requested host name '127.0.0.1'.
HTTP request sent, awaiting response... 404 Not Found
2023-11-03 21:25:56 ERROR 404: Not Found.
```
I have tried using <https://127.0.0.1:8443>, <https://127.0.0.1:8443/>, <https://127.0.0.1:8443/ui>, <https://127.0.0.1:8443/ui/>, <https://127.0.0.1:8443/web>, <https://127.0.0.1:8443/web/>, <https://127.0.0.1:8443/dashboard>, <https://127.0.0.1:8443/dashboard/> but they all return 404.

I have also tried with all 8 possible unique combinations of the boolean values listed below, but they all return 404.

```yml
    networkPolicy:
      enabled: true
      ingressDenyAll: true
  ingress:
    enabled: true
    hosts:
      - localhost
```

I have also tried using port-forward `svc/kubernetes-dashboard-nginx-controller-admission` instead of `svc/kubernetes-dashboard-nginx-controller`  but I still get 404.

Any suggestion of what could be wrong with my approach?

**Comment by ahoehma** [comment]
I have more or less the same setup ...

```yaml
cert-manager:
  enabled: false 
nginx:
  enabled: false
metrics-server:
  enabled: false
app:
  ingress:
    enabled: false 
```

$k get pods,svc,ingress -n kubernetes-dashboard
NAME                                                       READY   STATUS    RESTARTS   AGE
pod/kubernetes-dashboard-api-7cc955bcdd-b7kzn              1/1     Running   0          14h
pod/kubernetes-dashboard-metrics-scraper-556c9dd79-n7z6q   1/1     Running   0          14h
pod/kubernetes-dashboard-web-795dc7c5dc-2tsrh              1/1     Running   0          14h

NAME                                           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
service/kubernetes-dashboard-api               ClusterIP   XXXXX    <none>        9000/TCP   14h
service/kubernetes-dashboard-metrics-scraper   ClusterIP   XXXXX        <none>        8000/TCP   14h
service/kubernetes-dashboard-web               ClusterIP   XXXXX       <none>        8000/TCP   14h

I can only got something working with "kubectl proxy" and then I can open

http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/http:kubernetes-dashboard-web:8000/proxy/

But then the page keeps blank and I see a lot of console errors like

404 http://localhost:8001/api/v1/namespaces/kubernetes-dashboard/services/http:kubernetes-dashboard-web:8000/proxy/api/v1/login/status

I believe this is because of the split between web and api services. 

No glue how to handle this?! :-(

**Comment by comminutus** [comment]
i get the same problem. I'm trying to use it with `kubectl port-forward` and no ingress

**Comment by comminutus** [comment]
@Stolz can you change this to an issue instead of a Q&A?

**Comment by DeadGuruu** [comment]
Same problem here
